tools/python/openlr/quality.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import xml.etree.ElementTree as ET
+import operator
+
+from itertools import islice
+from collections import namedtuple
+from math import sin, cos, atan2, radians
+
+LatLon = namedtuple('LatLon', 'lat, lon')
+
+def distance(x, y):
+    """Implements https://en.wikipedia.org/wiki/Haversine_formula.
+
+    >>> int(distance(LatLon(55.747043, 37.655554), LatLon(55.754892, 37.657013)))
+    875
+    >>> int(distance(LatLon(60.013918, 29.718361), LatLon(59.951572, 30.205536)))
+    27910
+    """
+
+    φ1, φ2 = map(radians, [x.lat, y.lat])
+    λ1, λ2 = map(radians, [x.lon, y.lon])
+    Δφ = φ2 - φ1
+    Δλ = λ2 - λ1
+    a = sin(Δφ/2)**2 + cos(φ1) * cos(φ2) * sin(Δλ/2)**2
+    R = 6356863
+    return 2 * R * atan2(a**0.5, (1 - a)**0.5)
+
+def polyline_length(ps):
+    if len(ps) < 2:
+        return 0.0
+    return sum(
+        distance(ps[i], ps[i - 1])
+        for i in range(1, len(ps))
+    )
+
+def common_subpath(l1, l2, eq=operator.eq):
+    prefs_len = [
+        [0 for _ in range(len(l2) + 1)]
+        for _ in range(len(l1) + 1)
+    ]
+    for i in range(1, len(l1) + 1):
+        for j in range(1, len(l2) + 1):
+            if eq(l1[i - 1], l2[j - 1]):
+                prefs_len[i][j] = prefs_len[i - 1][j - 1] + 1
```

Comment:
Looks like you need to store sum of distances in prefs_len, not number of segments, right?

Reply:
Done.